File: webApi/Models/Product.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace webApi.Models
{
    // Represents a product in the system
    public class Product
    {
        // Unique identifier for the product
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        // Name of the product
        public string ProductName { get; set; }

        // URL or path to the product image
        public string ProductImage { get; set; }

        // Detailed description of the product
        public string ProductDescription { get; set; }

        // Price of the product
        public decimal ProductPrice { get; set; }

        // Rating of the product (e.g., out of 5)
        public decimal ProductRating { get; set; }

        // Category to which the product belongs
        public string CategoryName { get; set; }

        // Amount of product available in stock
        public int ProductStock { get; set; }

        // Indicates if the product is active (available for sale)
        public bool IsActive { get; set; }

        // Unique identifier for the vendor supplying the product
        [BsonRepresentation(BsonType.ObjectId)]
        public string VendorId { get; set; }

        // Name of the vendor supplying the product
        public string VendorName { get; set; }
    }
}

File: EADmobile/app/src/main/java/com/example/ead_mobile/model/Product.java
package com.example.ead_mobile.model;


public class Product {
    private String productId;
    private String productName;
    private String productImage;
    private String productDescription;
    private double productPrice;
    private double productRating;
    private String categoryName;
    private int productStock;
    private boolean isActive;
    private String vendorId;

    // Constructor
    public Product(String productId, String productName, String productImage, String productDescription,
                   double productPrice, double productRating, String categoryName, int productStock,
                   boolean isActive, String vendorId) {
        this.productId = productId;
        this.productName = productName;
        this.productImage = productImage;
        this.productDescription = productDescription;
        this.productPrice = productPrice;
        this.productRating = productRating;
        this.categoryName = categoryName;
        this.productStock = productStock;
        this.isActive = isActive;
        this.vendorId = vendorId;
    }

    // Getters
    public String getProductId() { return productId; }
    public String getProductName() { return productName; }
    public String getProductImage() { return productImage; }
    public String getProductDescription() { return productDescription; }
    public double getProductRating() { return productRating; }

    public double getProductPrice() {
        return productPrice;
    }

    public String getCategoryName() { return categoryName; }
    public int getProductStock() { return productStock; }
    public boolean isActive() { return isActive; }
    public String getVendorId() { return vendorId; }
}
